clientapi/routing/register.go

```diff
@@ -328,7 +328,18 @@ func UserIDIsWithinApplicationServiceNamespace(
 	userID string,
 	appservice *config.ApplicationService,
 ) bool {
+
+	var local, _, err = gomatrixserverlib.SplitID('@', userID)
+	if err != nil {
+		// Not a valid userID
+		return false
+	}
+
 	if appservice != nil {
+		if appservice.SenderLocalpart == local {
```

Comment:
Thought: We should check that the domain matches the current server. Is there an easy way to get the value of domain of the server?

Reply:
Added a check.